File: code_samples/haskell/design_patterns_decorator.hs
-- Design Patterns: Decorator
-- AI/ML Training Sample

module Decorator where

data Decorator = Decorator {
    getData :: String
} deriving (Show, Eq)

process :: Decorator -> String -> Decorator
process obj input = obj { getData = input }

validate :: Decorator -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Decorator { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)

File: code_samples/csharp/design_patterns_decorator.cs
using System;

/// <summary>
/// Design Patterns: Decorator
/// AI/ML Training Sample
/// </summary>
public class Decorator
{
    private string data;
    
    public Decorator()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Decorator();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}
